### ISSUE TITLE
Unexpected error occured. enum can't contains StrWhiteSpace name.

### ISSUE THREAD

**Issue by vvakame** [microsoft/TypeScript#1144]
I think this code is correct. but tsc 1.3.0 and master/HEAD not allowed. [Playground](http://goo.gl/fBwt1a)

```
enum Sample {
  '\t', // An enum number cannot have a numeric name.
  ' ', // An enum number cannot have a numeric name.
  '\n', // An enum number cannot have a numeric name.
  '                         ' // An enum number cannot have a numeric name.
}
```

https://github.com/Microsoft/TypeScript/blame/v1.3/src/compiler/checker.ts#L4094
http://www.ecma-international.org/ecma-262/5.1/#sec-15.1.2.4
http://www.ecma-international.org/ecma-262/5.1/#sec-9.3.1

I think `isNumericName` check argument is `StrNumericLiteral` instead of `StringNumericLiteral`.
I want to make pull request, but I can't come up come up with a fast way...

https://github.com/borisyankov/DefinitelyTyped/pull/3135
